### PR TITLE
Refactor directory structure 

### PR DESCRIPTION
**Pull request by mareklibra**
Depends on:
- [x] #678 Move cim components to the lib
- [x] https://github.com/openshift-assisted/assisted-ui-lib/pull/688

Related to:
- [ ] https://github.com/openshift-assisted/assisted-ui/pull/456
- [ ] https://github.com/rawagner/dynamic-cim/pull/32
- [ ] uhc-portal MR 2631

TODO:
- [x] The CIM tree
- [x] fix for uhc-portal - (MR 2631 ready, blocked on ai-ui-lib release)
- [x] resolve dual-export (index-ocm, index-cim)
- [x] Review what else should be trasnfered from OCM to either common or CIM
- [ ] verify Disk Role Selection (modified, might be broken)
- [x] rethink code around passing `onAdditionalNtpSource` and `AdditionalNTPSourcesDialogToggleComponent` - it's a mess recently
- [x] do print ai-ui-lib version in both CIM and OCM flows

### CODE REVIEW

File: src/cim/components/ClusterDeployment/ClusterDeploymentNetworking.tsx [hunk 1]
```diff
@@ -17,12 +19,19 @@ const ClusterDeploymentNetworking: React.FC<
...
 
+  const CimNetworkingStatus: React.FC<HostNetworkingStatusComponentProps> = React.useMemo(
+    () => (props) => <NetworkingStatus cluster={cluster} {...props} />,
+    [cluster],
+  );
+  const AdditionalNTPSourcesDialogToggleWithCluster: React.FC = React.useMemo(
+    () => () => <AdditionalNTPSourcesDialogToggle cluster={cluster} />,
+    [cluster],
+  );
+
```
Comment (rawagner) [inline src/cim/components/ClusterDeployment/ClusterDeploymentNetworking.tsx:30]: `useCallback` would be nicer

File: src/cim/components/ClusterDeployment/NetworkingStatus.tsx [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+const NetworkingStatus: React.FC<HostNetworkingStatusComponentProps & { cluster: Cluster }> = ({
+  cluster,
+  ...props
+}) => {
+  const networkingStatus: Host['status'] = 'discovering'; // TODO(mlibra)
+  const validationsInfo: ValidationsInfo = {};
+  const sublabel = undefined;
+
+  const AdditionalNTPSourcesDialogToggleWithCluster: ValidationInfoActionProps['AdditionalNTPSourcesDialogToggleComponent'] = React.useMemo(
```
Comment (rawagner) [inline src/cim/components/ClusterDeployment/NetworkingStatus.tsx:15]: useCallback

File: src/cim/components/common/ValidatedConditionAlert.tsx [hunk 3]
```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+import { Alert, AlertVariant } from '@patternfly/react-core';
+import { StatusCondition } from '../../types';
+
+type ValidationAlertProps = {
+  condition: StatusCondition<'Validated'>;
+};
+
+const ValidatedConditionAlert = ({ condition }: ValidationAlertProps) => {
```
Comment (rawagner) [inline src/cim/components/common/ValidatedConditionAlert.tsx:9]:
```suggestion
const ValidatedConditionAlert: React.FC<ValidationAlertProps> = ({ condition }) => {
```

File: src/common/components/clusterDetail/ClusterProgress.tsx [hunk 4]
```diff
@@ -168,9 +168,10 @@ const getFinalizingStatusIcon = (cluster: Cluster) => {
 
 type FinalizingProgressProps = {
   cluster: Cluster;
+  onFetchEvents: EventListFetchProps['onFetchEvents'];
 };
 
-const FinalizingProgress = ({ cluster }: FinalizingProgressProps) => {
+const FinalizingProgress = ({ cluster, onFetchEvents }: FinalizingProgressProps) => {
```
Comment (rawagner) [inline src/common/components/clusterDetail/ClusterProgress.tsx:174]: not new, but still
```suggestion
const FinalizingProgress: React.FC<FinalizingProgressProps> = ({ cluster, onFetchEvents }) => {
```

**Comment by mareklibra** [comment]
Rebased

**Review by jtomasek — DISMISSED** [review]

**Comment by mareklibra** [comment]
Just rebased.

**Review by rawagner — APPROVED** [review]

**Pull request merged**